### PR TITLE
50 Add cross compilation ability

### LINKED ISSUE #50: Add scala cross-compilation for 2.11 and 2.12
Add scala cross-compilation for 2.11 and 2.12

Blocks #49

### PR DESCRIPTION
**Pull request by Zejnilovic**
Uses @wajda 's cross-compilation plugin adds 2.11 and 2.12 compilations

Fixes #50

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,8 +91,8 @@ indepedently.
 ```xml
 <dependency>
       <groupId>za.co.absa</groupId>
-      <artifactId>atum</artifactId>
```
Comment (yruslan) [inline README.md:94]: Maybe README should contain references to both 2.11 and 2.12

**Review by yruslan — APPROVED** [review]

**Review by dk1844 — COMMENTED** [review]
Will it play nice with the release plugin? 
We probably want to release both 2.11 & 2.12 versions with the same version number, so at-once somehow. I wonder if we can.

File: pom.xml [hunk 2]
```diff
@@ -102,6 +102,17 @@
...
+
+        <scala_2.11.version>2.11.12</scala_2.11.version>
+        <scala_2.12.version>2.12.12</scala_2.12.version>
+
+        <!-- Controlled by `scala-cross-build` plugin -->
+        <scala.version>2.11.12</scala.version>
+        <scala.binary.version>2.11</scala.binary.version>
+
         <!-- Frameworks and libraries -->
         <json4s.version>3.5.3</json4s.version> <!-- This version is set to be compatible with Spark 2.4.5 -->
```
[the remark this answers is not in the document]
Comment (Zejnilovic) [reply pom.xml:117]: Yeah, I am trying to take it one problem at a time. Also, I think this was a problem from 2.2 to 2.4. Version 3 could be fine.

**Comment by Zejnilovic** [comment]
> Will it play nice with the release plugin?
> We probably want to release both 2.11 & 2.12 versions with the same version number, so at-once somehow. I wonder if we can.

This is a setup used by our commons so it should

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (wajda) [reply pom.xml:117]: Spline agent is being tested with Spark 3.0. AFAIK our commons Json4s wrapper works fine in all versions

**Comment by wajda** [comment]
> Will it play nice with the release plugin?
> We probably want to release both 2.11 & 2.12 versions with the same version number, so at-once somehow. I wonder if we can.

If we're talking about `release:prepare` then yes, it will play nicely with it. The trick here is to use `release` plugin to __only__ take care about versioning (i.e. increment version, create tag in VCS etc). For that you don't care about scala version simply because you don't compile/deploy/publish anything, you only work with the codebase itself, which is the same both all scala binary versions. Even though technically release plugin needs to compile the project, it's enough for the build to pass for any (default) scala ver. The assumption is that the actual testing per every Scala ver is done from the same commit in the CI pipeline before conducting the release stage.

**Review by dk1844 — APPROVED** [review]

File: build-all.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+#
+# THIS SCRIPT IS INTENDED FOR LOCAL DEV USAGE ONLY
+#
+
+BASE_DIR=$(dirname "$0")
+
+cross_build() {
+  SCALA_VER=$1
+  echo "Building with Scala $SCALA_VER"
+  find $BASE_DIR/target/* -type d -exec rm -rf {} \;
```
Comment (Zejnilovic) [inline build-all.sh:26]: update for the multimodule needed

**Review by AdrianOlosutean — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -87,12 +87,20 @@ indepedently.
 ## Usage
 
 ### Coordinate for Maven POM dependancy
-
+For Scala 2.11
```
Comment (benedeki) [inline README.md:90]: **Tiny:**
```suggestion
For project using Scala 2.11
```
I think this make it cleaner what the example intends to show.

File: README.md [hunk 5]
```diff
@@ -87,12 +87,20 @@ indepedently.
...
+```xml
+<dependency>
+      <groupId>za.co.absa</groupId>
+      <artifactId>atum_2.11</artifactId>
+      <version>3.2.0</version>
+</dependency>
+```
+For Scala 2.12
```
Comment (benedeki) [inline README.md:98]:
```suggestion
For project using Scala 2.12
```
In accordance with the previous suggestion.

**Review by benedeki — APPROVED** [review]

**Pull request merged**